packages/web-react/src/components/Card/useCardStyleProps.ts

```diff
@@ -0,0 +1,63 @@
+import classNames from 'classnames';
+import { useAlignmentClass, useClassNamePrefix } from '../../hooks';
+import { CardAlignmentXType, DirectionDictionaryType, HorizontalReversedType, SizesDictionaryType } from '../../types';
+import { kebabCaseToCamelCase } from '../../utils';
+
+export interface UseCardStyleProps {
+  alignmentX?: CardAlignmentXType;
+  direction?: DirectionDictionaryType | HorizontalReversedType;
+  hasFilledHeight?: boolean;
+  isBoxed?: boolean;
+  isExpanded?: boolean;
+  isHeading?: boolean;
+  size?: SizesDictionaryType | 'auto';
+}
+
+export interface UseCardStylePropsReturn {
+  /** className props */
+  classProps: {
+    body: string;
+    eyebrow: string;
+    footer: string;
+    media: string;
+    root: string;
+    title: string;
+  };
+}
+
+export function useCardStyleProps(props?: UseCardStyleProps): UseCardStylePropsReturn {
+  const { direction, isBoxed, isExpanded, alignmentX, size, isHeading, hasFilledHeight } = props || {};
+  const cardClass = useClassNamePrefix('Card');
+  const isBoxedClass = `${cardClass}--boxed`;
+  const directionClass = direction ? `${cardClass}--${kebabCaseToCamelCase(direction)}` : '';
+  const mediaSizeClass = size ? `${cardClass}Media--${size}` : '';
+  const mediaIsExpandedClass = `${cardClass}Media--expanded`;
+  const mediaHasFilledHeightClass = `${cardClass}Media--filledHeight`;
+
+  const rootClasses = classNames(cardClass, directionClass, {
+    [isBoxedClass]: isBoxed,
+  });
+  const mediaClasses = classNames(`${cardClass}Media`, mediaSizeClass, {
```

Comment:
I would define all these classes in the section above.
```
const mediaClass = `${cardClass}Media`;
```
```suggestion
  const mediaClasses = classNames(mediaClass, mediaSizeClass, {
```

Reply:
the same for the rest